elide-async/src/main/java/com/yahoo/elide/async/service/thread/AsyncQueryThread.java

```diff
@@ -62,10 +69,31 @@ public AsyncAPIResult call() throws URISyntaxException, NoHttpResponseException
         queryResult.setCompletedOn(new Date());
         queryResult.setResponseBody(response.getBody());
         queryResult.setContentLength(response.getBody().length());
-        //TODO Add recordcount to queryResultObj
+        queryResult.setRecordCount(calculateRecordCount(response, queryObj.getQueryType()));
         return queryResult;
     }
 
+    /**
+     * This method calculates the number of records from the response.
+     * @param response is the ElideResponse
+     * @param queryType is the query type (GraphQL or JSON).
+     * @return Record Count
+     * @throws IOException Exception thrown by JsonPath
+     */
+    protected Integer calculateRecordCount(ElideResponse response, QueryType queryType)
+            throws IOException {
+        Integer count = null;
+        if (response.getResponseCode() == 200) {
+            if (queryType.equals(QueryType.GRAPHQL_V1_0)) {
```

Comment:
The code has these query type checks sprinkled in many different places.  
This suggests to me that we ought to have an interface (and concrete functions) with a set of functions that we need each query type to perform.